### PR TITLE
JA-P5 integration for onboarding to dashboard

### PR DESCRIPTION
**Pull request by jagadeepavula**
[PC-159](https://makeitmvp.atlassian.net/browse/P5AP-159)

## Type of change

- [ ] Bug fix
- [X] New feature
- [X] Refactoring

## Description

- Changes made?
integrated the firestore data with the dashboard page
- Reason for changes?
needed integration to fetch data from backend
- Behavior before?
Fetched data from MOCK_CHALLENGES file
- Behavior after?
now fetches it from firestore challenges collection. Also integrated filter board. User can update their change in board for results. 
- How did you test?
Locally and console logged to check if I'm getting the required information in each page and code.

## Checklist

- [ ] I have performed a self-review of my code.
- [X] I have manually tested the new feature and found it to be working as expected.
- [ ] All commits in this PR are related to the issue.
- [X] Base branch for this PR is set to `develop`.
- [X] JIRA ticket # and my initials are noted in this PR's name (format example: #57-EB Add Pull Request template)

**Reminder**: every PR needs at least 1 peer approval before merging to the `develop` branch. Once a main ticket has been completed, along with all subtasks, associated code can be merged to `main`.

### CODE REVIEW

File: src/components/P5AP_ChallengePage/ChallengePage.jsx [hunk 1]
```diff
@@ -1,34 +1,103 @@
...
 import ChallengeCard from "../ChallengeCard/ChallengeCard";
 import DashboardNavbar from "../DashboardNavbar/DashboardNavbar";
 import filterIcon from "../../assets/images/filterIcon.svg";
 import FilterBoard from "../P5AP_FilterBoard/FilterBoard";
+import { PageContext } from "../../contexts/PageContext";
 import "./ChallengePage.scss";
 
 const ChallengePage = () => {
+  const { userFeedback } = useContext(PageContext);
+  console.log("ChallengePage mounted, userFeedback:", userFeedback);
```
Comment (TimWallaceDev) [inline src/components/P5AP_ChallengePage/ChallengePage.jsx]: I noticed there are quite a few console.log left from testing. These should be removed

File: src/components/P5AP_ChallengePage/ChallengePage.jsx [hunk 2]
```diff
@@ -1,34 +1,103 @@
...
-        selectedDifficulties.includes(challenge.difficulty)) &&
-      (selectedTechnologies.length === 0 ||
-        selectedTechnologies.includes(challenge.category))
-  );
+  useEffect(() => {
+    const fetchChallenges = async () => {
+      setIsLoading(true);
+      try {
+        const fetchedChallenges = await readData("Challenges");
+        console.log("Fetched challenges:", fetchedChallenges);
```
Comment (TimWallaceDev) [inline src/components/P5AP_ChallengePage/ChallengePage.jsx]: console.log

File: src/components/P5AP_ChallengePage/ChallengePage.jsx [hunk 3]
```diff
@@ -1,34 +1,103 @@
...
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchChallenges();
+  }, []);
+
+  useEffect(() => {
+    console.log("Setting initial filters. UserFeedback:", userFeedback);
```
Comment (TimWallaceDev) [inline src/components/P5AP_ChallengePage/ChallengePage.jsx]: console.log

File: src/components/P5AP_ChallengePage/ChallengePage.jsx [hunk 4]
```diff
@@ -1,34 +1,103 @@
...
+    }
+    console.log("Filters set:", {
+      industry: selectedIndustry,
+      difficulties: selectedDifficulties,
+      technologies: selectedTechnologies,
+    });
+  }, [userFeedback]);
+
+  useEffect(() => {
+    console.log("Current filter settings:");
```
Comment (TimWallaceDev) [inline src/components/P5AP_ChallengePage/ChallengePage.jsx]: console.log

File: src/components/P5AP_ChallengePage/ChallengePage.jsx [hunk 5]
```diff
@@ -1,34 +1,103 @@
...
+  const filteredChallenges = challenges.filter((challenge) => {
+    const industryMatch = true; // Always true because "Development" is always included
+    const difficultyMatch =
+      selectedDifficulties.length === 0 ||
+      selectedDifficulties.includes(challenge.difficulty);
+    const technologyMatch =
+      selectedTechnologies.length === 0 ||
+      selectedTechnologies.some((tech) => challenge.category.includes(tech));
+
+    console.log(
```
Comment (TimWallaceDev) [inline src/components/P5AP_ChallengePage/ChallengePage.jsx]: console.log

File: src/components/P5AP_ChallengePage/ChallengePage.jsx [hunk 6]
```diff
@@ -1,34 +1,103 @@
...
-    setSelectedIndustry(filters.industry);
-    setSelectedDifficulties(filters.difficulty);
-    setSelectedTechnologies(filters.technology);
+    setSelectedIndustry(["Development", ...(filters.industry || [])]);
+    setSelectedDifficulties(filters.difficulty || []);
+    setSelectedTechnologies(filters.technology || []);
     setIsFilterBoardOpen(false);
   };
 
+  if (isLoading) {
```
Comment (TimWallaceDev) [inline src/components/P5AP_ChallengePage/ChallengePage.jsx:67]: Great handling of the loading state

File: src/components/P5AP_ChallengePage/ChallengePage.jsx [hunk 7]
```diff
@@ -60,19 +128,22 @@ const ChallengePage = () => {
           ))}
         </div>
 
-        {/* Challenge Cards */}
         <div className="card-container">
-          {filteredChallenges.map((challenge, index) => (
-            <ChallengeCard key={index} challenge={challenge} />
-          ))}
+          {filteredChallenges.length > 0 ? (
```
Comment (TimWallaceDev) [inline src/components/P5AP_ChallengePage/ChallengePage.jsx:106]: Great use of a ternary operator

File: src/pages/OnboardingPage/OnboardingPage3.jsx [hunk 8]
```diff
@@ -7,7 +7,21 @@ import loadingActiveIcon from "../../assets/images/loadingActiveIcon.svg";
...
+  const {
+    handleNext,
+    handleBack,
+    progressBarIndex,
+    updateUserFeedback,
+    userFeedback,
+  } = useContext(PageContext);
+
+  const handleSkillSelection = (skill) => {
+    console.log("Skill selected:", skill);
```
Comment (TimWallaceDev) [inline src/pages/OnboardingPage/OnboardingPage3.jsx]: console.log

**Review by TimWallaceDev — APPROVED** [review]
I tried this branch on my local machine and can confirm it is working as expected

**Comment by jagadeepavula** [comment]
I removed all the console logs

**Comment by TimWallaceDev** [comment]
Should be good to merge

**Pull request merged**